pkg/handler3/handler.go

```diff
@@ -117,7 +133,11 @@ func (o *OpenAPIService) getSingleGroupBytes(getType string, group string) ([]by
 		specBytes, etag, err := v.jsonCache.Get()
 		return specBytes, etag, v.lastModified, err
 	} else if getType == subTypeProtobuf {
-		specPb, etag, err := v.pbCache.Get()
+		_, etag, err := v.jsonCache.Get()
+		if err != nil {
+			return nil, "", v.lastModified, err
+		}
+		specPb, _, err := v.pbCache.Get()
```

Comment:
I think we're doing the wrong thing with the lazy caching base-code. If we don't like how the etag is computed (since clearly we don't want one per type anymore), we should change the underlying lazy-caching code.

Reply:
Note that this probably also applies to other things below.

Reply:
I sort of fixed this by allowing the user to pass in an optional ETag in the New function when initializing the cache. If the ETag is empty then we'll compute it. Let me know your thoughts.

Reply:
Created a new cache for ETag and decoupled from byteCache.